lib/huginn_acumen_order_agent/concerns/invoice_detail_query_concern.rb

```diff
@@ -41,9 +41,10 @@ def process_invoice_detail_response(raw_data)
         results << mapped_item
       rescue => error
         issue_error(AcumenOrderError.new(
+          error[:status],
```

Comment:
This won't work.

This assumes that `error` is a hash -- it's not. It's an Error object. The correct syntax would be `error.status`
However, that is also incorrect as it assumes that `error` will always have a field called `status`, and we don't know that.

If we're going to set a `status` on the `AcumenAgentError`, then we need to somehow get a relevant status -- 404, 422, 503, etc.

If we don't have enough to identify which status code to use, then just use `500`.

Reply:
@mrogers-5s I think I fixed this as well, let me know what you think.